great_expectations/validator/validator.py

```diff
@@ -360,6 +375,298 @@ def inst_expectation(*args, **kwargs):
 
         return inst_expectation
 
+    def _build_expectation_configuration(
+        self,
+        expectation_type: str,
+        expectation_kwargs: dict,
+        meta: dict,
+        expectation_impl: "Expectation",  # noqa: F821
+    ) -> ExpectationConfiguration:
+        success_keys: Tuple[str] = (
+            expectation_impl.success_keys
+            if hasattr(expectation_impl, "success_keys")
+            else None
+        ) or tuple()
+        arg_keys: Tuple[str] = (
+            expectation_impl.arg_keys if hasattr(expectation_impl, "arg_keys") else None
+        ) or tuple()
+        runtime_keys: Tuple[str] = (
+            expectation_impl.runtime_keys
+            if hasattr(expectation_impl, "runtime_keys")
+            else None
+        ) or tuple()
+        # noinspection PyTypeChecker
+        override_keys: Tuple[str] = success_keys + arg_keys + runtime_keys
+
+        key: str
+        value: Any
+        expectation_kwargs_overrides: dict = {
+            key: value
+            for key, value in expectation_kwargs.items()
+            if key in override_keys
+        }
+
+        auto: Optional[bool] = expectation_kwargs.get("auto")
+        profiler_config: Optional[
+            Union[
+                RuleBasedProfilerConfig,
+                Dict[
+                    str,
+                    Union[
+                        str,
+                        float,
+                        Optional[Dict[str, Any]],
+                        Optional[Dict[str, Dict[str, Any]]],
+                    ],
+                ],
+            ]
```

Comment:
Do we have a cleaner approach to annotate the dict representation of configs? Can't immediately think of one but I'd really like to avoid super complex types like this if possible (I realize this isn't entirely feasible here so do you what you think is best).

Reply:
@cdkini Perhaps for a later refactor (it is just me being explicit here; we could of course just say `dict`, but I wanted to be more descriptive so that in a future refactor we will immediately know "the type that is actually there".  Thanks.

Reply:
Please remove the Dict here, we don't need to support both.

Reply:
Can we now replace this type hint with `Optional[RuleBasedProfilerConfig]`?